### PR TITLE
Enumerated the possible build phase states in the comment

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Stop forcing a re-build when updating generated completions

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

Enumerated the possible build phase states in the comment

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
[test]

@bparees @csrwng  this gets `oc explain build.status.phase` to be useful ... also the second commit gets rid of a minute on our build times ....

File: hack/update-generated-completions.sh [hunk 1]
```diff
@@ -4,9 +4,6 @@
 
 source "$(dirname "${BASH_SOURCE}")/lib/init.sh"
 
-# ensure we have the latest compiled binaries
-"${OS_ROOT}/hack/build-go.sh"
```
Comment (bparees) [inline hack/update-generated-completions.sh]: this should be basically free if we've already done a build anyway.  and if you haven't done a build, the completions you generate are going to be stale/wrong, no?

**Comment by csrwng** [comment]
Please also update https://github.com/openshift/origin/blob/master/pkg/build/api/types.go#L231

File: hack/update-generated-completions.sh (hunk 1 above)
Comment (stevekuznetsov) [inline hack/update-generated-completions.sh]: It takes 1minute. We do not do this in _any_ script and this one doesn't get to be special.

Comment (stevekuznetsov) [inline hack/update-generated-completions.sh]: To be more clear -- `make update` depends on `make build`. Update does updates. Build does builds. This has been our convention for the entire lifetime of this project.

Comment (bparees) [inline hack/update-generated-completions.sh]: k

**Comment by stevekuznetsov** [comment]
This doesn't look good...
```
[INFO] Dumping container logs to /tmp/openshift/conformance/logs/containers
Error response from daemon: devmapper: Unknown device 3bbbada5bb62f008dcfe5ed446735673e84eb50aff3ecf350f0b9b62dc4053f7
json: cannot unmarshal array into Go value of type types.ContainerJSON
Error: No such image, container or task: 2b38f4229f9d
json: cannot unmarshal array into Go value of type types.ContainerJSON
Error response from daemon: devmapper: Unknown device 25ce75cc2caaf86484e829f3ae46a7024c318d8704323271c76bd1cf80788721
```

**Comment by stevekuznetsov** [comment]
@openshift-bot, the [last build](https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin/626/) failed from the following flakes:
 - [integration](https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_extended_conformance_install/870/consoleFull#15372050265898c58db7602c31c0eab717) failed on: https://github.com/openshift/origin/issues/13672

[test]

**Comment by stevekuznetsov** [comment]
@bparees any comments?

**Comment by bparees** [comment]
lgtm [merge]

**Pull request merged**